pkg/os/exec.go

```diff
@@ -51,26 +51,35 @@ func RunWithPrivilege(reason string, cmdAndArgs ...string) (string, string, erro
 	return run(sudo, cmdAndArgs, map[string]string{})
 }
 
+var defaultLocaleEnv = map[string]string{"LC_ALL": "C", "LANG": "C"}
+
 func RunWithDefaultLocale(command string, args ...string) (string, string, error) {
-	return RunWithDefaultLocaleAndEnv(command, args, map[string]string{})
+	return run(command, args, defaultLocaleEnv)
 }
 
 func RunWithDefaultLocalePrivate(command string, args ...string) (string, string, error) {
-	return RunWithDefaultLocalePrivateAndEnv(command, args, map[string]string{})
+	return runPrivate(command, args, defaultLocaleEnv)
 }
 
-func RunWithDefaultLocaleAndEnv(command string, args []string, extraEnv map[string]string) (string, string, error) {
-	env := map[string]string{"LC_ALL": "C", "LANG": "C"}
-	for k, v := range extraEnv {
-		env[k] = v
-	}
-	return run(command, args, env)
+type CommandRunner interface {
+	Run(command string, args ...string) (string, string, error)
+	RunPrivate(command string, args ...string) (string, string, error)
+	RunPrivileged(reason string, cmdAndArgs ...string) (string, string, error)
 }
+type localRunner struct{}
 
-func RunWithDefaultLocalePrivateAndEnv(command string, args []string, extraEnv map[string]string) (string, string, error) {
-	env := map[string]string{"LC_ALL": "C", "LANG": "C"}
-	for k, v := range extraEnv {
-		env[k] = v
-	}
-	return runPrivate(command, args, env)
```

Comment:
These both removed function is used to get `oc config view` because `oc` doesn't respect `--kubeconfig` option when you try to merge the 2 different kubeconfig, this use case is not required since it we are using oc package now. I think in the future this might be again need refactor when some command require and env variable set before executing. I am not sure we should have it right now or again do refactor around when this requirement comes.

Reply:
Yup, it can always be readded if/when needed